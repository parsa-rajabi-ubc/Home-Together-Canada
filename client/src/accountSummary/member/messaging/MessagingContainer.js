/**
 *
 * @Author:     Jeff Hatton
 * @Created:    2021.3.04
 *
 * @Description: Member Messaging Conversation Card List Container
 *
 */
import React, {useEffect, useState} from 'react';
import Confirmation from "../../../common/listings/Confirmation";
import SelectConversation from "./SelectConversation";
import MessageService from "../../../services/MessageService";

const MESSAGE = {
    NO_RESULTS: "No results found"
}

function MessagingContainer(){
    const [messageData,setMessageData] = useState([]);
    const [myUserId,setMyUserId] = useState(-1);
    const [loadingMessageData, setLoadingMessageData] = useState(true);
    const [loadingUid, seUidLoading] = useState(true);

    // TODO: Save the user's uid in the redux store.

    useEffect(() => {
        MessageService.getCurrentRegisteredUId()
            .then(res => res.json())
            .then(data => {
                setMyUserId(data.uid);
                seUidLoading(false);
            })
            .catch(err => {
                alert('error getting message: ' + err.message);
                seUidLoading(false);
            });
    }, [])

    useEffect(() => {
        MessageService.getAllMessagesForOneUser()
            .then(res => res.json())
            .then(data => {
                setMessageData(data.message);
                setLoadingMessageData(false);
            })
            .catch(err => {
                alert('error getting message: ' + err.message);
                setLoadingMessageData(false);
            });
    }, [])

    // TODO: refactor to use this function in the useEffect above
    const getAllMessagesForUser = () => {
            MessageService.getAllMessagesForOneUser()
                .then(res => res.json())
                .then(data => {
                    setMessageData(data.message);
                })
                .catch(err => {
                    alert('error getting message: ' + err.message);
                });
        };

    return (
        <div>
            {(!loadingMessageData && !loadingUid ) &&
                <div className={"m-6"}>
                    {(!messageData.length)
                        ? <Confirmation displayButton={false} errorColor={true} message={MESSAGE.NO_RESULTS}/>
                        : <SelectConversation
                            messageData={messageData}
                            myUserId={myUserId}
                            newMessageAddedCallback={getAllMessagesForUser}
                        />
                    }
                </div>
            }
        </div>
    );
}

export default MessagingContainer;