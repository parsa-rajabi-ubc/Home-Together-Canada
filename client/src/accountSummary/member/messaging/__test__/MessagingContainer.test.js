/**
 * @Author:     Jeff Hatton
 * @Created:    2021.3.06
 *
 * @Description: Conversation List Snapshot Test
 *
 */
import React from 'react';
import MessagingContainer from "../MessagingContainer";
import ShallowRenderer from "react-test-renderer/shallow";

jest.mock("uuid", () => ({
            v4: () => "00000000-0000-0000-0000-000000000000"
        }
    )
);
describe('MessagingContainer', () => {
    describe('Snapshot test', () => {
        it('should match snapshot test if listing is NOT empty for userName', () => {
            //given

            //when
            const renderer = new ShallowRenderer();
            renderer.render(<MessagingContainer/>);
            const result = renderer.getRenderOutput();
            //then
            expect(result).toMatchSnapshot();
        });
    });
});