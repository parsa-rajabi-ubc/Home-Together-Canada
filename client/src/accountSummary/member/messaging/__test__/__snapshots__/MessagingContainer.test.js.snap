// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagingContainer Snapshot test should match snapshot test if listing is NOT empty for userName 1`] = `
<div
  className="m-6"
>
  <SelectConversation
    messageData={
      Array [
        Object {
          "dateSent": "2021-01-12T02:42:39.005Z",
          "id": 1,
          "messageContent": "message content 1 all on different days and times",
          "receiverId": "otherUser1",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-01T02:42:39.005Z",
          "id": 2,
          "messageContent": "message content 2",
          "receiverId": "otherUser2",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-02-14T08:42:39.005Z",
          "id": 3,
          "messageContent": "message content 3",
          "receiverId": "otherUser3",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-12T02:42:39.005Z",
          "id": 4,
          "messageContent": "message content 4",
          "receiverId": "otherUser4",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2020-03-13T02:42:39.075Z",
          "id": 5,
          "messageContent": "message content 5",
          "receiverId": "otherUser5",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-02-17T02:43:39.005Z",
          "id": 6,
          "messageContent": "message content 6",
          "receiverId": "otherUser6",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-19T09:42:39.005Z",
          "id": 7,
          "messageContent": "message content 7",
          "receiverId": "otherUser7",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-11T02:42:39.005Z",
          "id": 8,
          "messageContent": "message content 8",
          "receiverId": "otherUser8",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-10T02:42:39.005Z",
          "id": 9,
          "messageContent": "message content 9 has timestamp before a smaller message id",
          "receiverId": "otherUser9",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-02-10T02:42:39.005Z",
          "id": 10,
          "messageContent": "message content 10",
          "receiverId": "otherUser10",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-14T02:42:39.005Z",
          "id": 11,
          "messageContent": "message content 11 all on different hours",
          "receiverId": "messageMember1",
          "senderId": "otherUser11",
        },
        Object {
          "dateSent": "2021-01-14T02:42:39.005Z",
          "id": 12,
          "messageContent": "message content 12",
          "receiverId": "messageMember",
          "senderId": "otherUser12",
        },
        Object {
          "dateSent": "2021-01-14T02:42:39.005Z",
          "id": 13,
          "messageContent": "message content 13",
          "receiverId": "messageMember1",
          "senderId": "otherUser13",
        },
        Object {
          "dateSent": "2021-01-14T02:42:39.777Z",
          "id": 14,
          "messageContent": "message content 14 other millisecond",
          "receiverId": "messageMember1",
          "senderId": "otherUser14",
        },
        Object {
          "dateSent": "2020-01-14T02:42:39.005Z",
          "id": 15,
          "messageContent": "message content 15 other year",
          "receiverId": "messageMember1",
          "senderId": "otherUser15",
        },
        Object {
          "dateSent": "2021-02-14T02:42:39.005Z",
          "id": 16,
          "messageContent": "message content 16 other month",
          "receiverId": "messageMember1",
          "senderId": "otherUser16",
        },
        Object {
          "dateSent": "2021-01-17T02:42:39.005Z",
          "id": 17,
          "messageContent": "message content 17 other day",
          "receiverId": "messageMember1",
          "senderId": "otherUser17",
        },
        Object {
          "dateSent": "2021-01-14T01:42:39.005Z",
          "id": 18,
          "messageContent": "message content 18 other hour",
          "receiverId": "messageMember1",
          "senderId": "otherUser18",
        },
        Object {
          "dateSent": "2021-01-14T02:47:39.005Z",
          "id": 19,
          "messageContent": "message content 19 other minute",
          "receiverId": "messageMember1",
          "senderId": "otherUser19",
        },
        Object {
          "dateSent": "2021-01-14T02:42:55.005Z",
          "id": 20,
          "messageContent": "message content 20 other second",
          "receiverId": "messageMember1",
          "senderId": "otherUser20",
        },
        Object {
          "dateSent": "2021-01-15T03:42:39.005Z",
          "id": 21,
          "messageContent": "message content 21",
          "receiverId": "otherUser1",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-16T04:42:39.005Z",
          "id": 22,
          "messageContent": "message content 22",
          "receiverId": "otherUser1",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-17T05:42:39.005Z",
          "id": 23,
          "messageContent": "message content 23",
          "receiverId": "messageMember1",
          "senderId": "otherUser1",
        },
        Object {
          "dateSent": "2021-01-18T06:42:39.005Z",
          "id": 24,
          "messageContent": "message content 24",
          "receiverId": "otherUser1",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-19T07:42:39.005Z",
          "id": 25,
          "messageContent": "message content 25",
          "receiverId": "messageMember1",
          "senderId": "otherUser1",
        },
        Object {
          "dateSent": "2021-01-20T08:42:39.005Z",
          "id": 26,
          "messageContent": "message content 26",
          "receiverId": "otherUser1",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-21T09:42:39.005Z",
          "id": 27,
          "messageContent": "message content 27",
          "receiverId": "otherUser1",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-22T10:42:39.005Z",
          "id": 28,
          "messageContent": "message content 28",
          "receiverId": "otherUser1",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-23T11:42:39.005Z",
          "id": 29,
          "messageContent": "message content 29",
          "receiverId": "otherUser1",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-24T12:42:39.005Z",
          "id": 30,
          "messageContent": "message content 30",
          "receiverId": "otherUser1",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-25T13:42:39.005Z",
          "id": 31,
          "messageContent": "message content 31",
          "receiverId": "otherUser1",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-26T14:42:39.005Z",
          "id": 32,
          "messageContent": "message content 32",
          "receiverId": "messageMember1",
          "senderId": "otherUser1",
        },
        Object {
          "dateSent": "2021-01-27T15:42:39.005Z",
          "id": 33,
          "messageContent": "message content 33",
          "receiverId": "otherUser1",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-28T16:42:39.005Z",
          "id": 34,
          "messageContent": "message content 34",
          "receiverId": "otherUser1",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-29T17:42:39.005Z",
          "id": 35,
          "messageContent": "message content 35",
          "receiverId": "otherUser1",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-30T18:42:39.005Z",
          "id": 36,
          "messageContent": "message content 36 all on different days and times",
          "receiverId": "otherUser1",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-02T02:42:39.005Z",
          "id": 37,
          "messageContent": "message content with lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots of padding 37",
          "receiverId": "otherUser2",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-03T02:42:39.005Z",
          "id": 38,
          "messageContent": "message content with lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots of padding 38",
          "receiverId": "messageMember1",
          "senderId": "otherUser2",
        },
        Object {
          "dateSent": "2021-01-04T02:42:39.005Z",
          "id": 39,
          "messageContent": "message content with lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots of padding 39",
          "receiverId": "otherUser2",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-05T02:42:39.005Z",
          "id": 40,
          "messageContent": "message content with lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots of padding 40",
          "receiverId": "messageMember1",
          "senderId": "otherUser2",
        },
        Object {
          "dateSent": "2021-01-06T02:42:39.005Z",
          "id": 41,
          "messageContent": "message content with lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots of padding 41",
          "receiverId": "otherUser2",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-07T07:42:39.005Z",
          "id": 42,
          "messageContent": "message content with lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots and lots of padding 42",
          "receiverId": "messageMember1",
          "senderId": "otherUser2",
        },
        Object {
          "dateSent": "2021-01-12T02:42:39.005Z",
          "id": 43,
          "messageContent": "message content 43 has same timestamp as message 1 and 44",
          "receiverId": "otherUser4",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-12T02:42:39.005Z",
          "id": 44,
          "messageContent": "message content 44 has same timestamp as message 1 and 43",
          "receiverId": "messageMember1",
          "senderId": "otherUser4",
        },
        Object {
          "dateSent": "2021-01-14T02:42:39.005Z",
          "id": 45,
          "messageContent": "message content 45 has same timestamp as message 2 and 46",
          "receiverId": "otherUser8",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-14T02:42:39.005Z",
          "id": 46,
          "messageContent": "message content 46 has same timestamp as message 2 and 45",
          "receiverId": "otherUser9",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-14T03:42:39.005Z",
          "id": 47,
          "messageContent": "message content 47",
          "receiverId": "otherUser11",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-14T04:42:39.005Z",
          "id": 48,
          "messageContent": "message content 48",
          "receiverId": "messageMember1",
          "senderId": "otherUser11",
        },
        Object {
          "dateSent": "2021-01-14T05:42:39.005Z",
          "id": 49,
          "messageContent": "message content 49",
          "receiverId": "otherUser11",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-14T06:42:39.005Z",
          "id": 50,
          "messageContent": "message content 50",
          "receiverId": "otherUser11",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-14T07:42:39.005Z",
          "id": 51,
          "messageContent": "message content 51 all on different hours",
          "receiverId": "messageMember1",
          "senderId": "otherUser11",
        },
        Object {
          "dateSent": "2021-01-14T02:42:39.005Z",
          "id": 52,
          "messageContent": "message content 52",
          "receiverId": "otherUser21",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-02-19T02:42:39.005Z",
          "id": 53,
          "messageContent": "message content 53",
          "receiverId": "otherUser22",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-14T02:42:39.005Z",
          "id": 54,
          "messageContent": "message content 54",
          "receiverId": "otherUser23",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-03-01T02:42:39.005Z",
          "id": 55,
          "messageContent": "message content 55",
          "receiverId": "otherUser24",
          "senderId": "messageMember1",
        },
        Object {
          "dateSent": "2021-01-14T02:42:39.005Z",
          "id": 56,
          "messageContent": "message content 56",
          "receiverId": "otherUser1",
          "senderId": "messageMember2",
        },
        Object {
          "dateSent": "2021-01-14T02:42:39.005Z",
          "id": 57,
          "messageContent": "message content 57",
          "receiverId": "otherUser1",
          "senderId": "messageMember3",
        },
        Object {
          "dateSent": "2021-01-14T02:42:39.005Z",
          "id": 58,
          "messageContent": "message content 58",
          "receiverId": "otherUser2",
          "senderId": "messageMember7",
        },
        Object {
          "dateSent": "2021-01-14T02:42:39.005Z",
          "id": 59,
          "messageContent": "message content 59",
          "receiverId": "otherUser2",
          "senderId": "otherUser1",
        },
        Object {
          "dateSent": "2021-01-14T02:42:39.005Z",
          "id": 60,
          "messageContent": "message content 60",
          "receiverId": "messageMember7",
          "senderId": "messageMember2",
        },
      ]
    }
    messageUser="messageMember1"
  />
</div>
`;
